### PR TITLE
[SYCL] Improve Doxygen documentation UX

### PR DESCRIPTION
**Pull request by alexbatashev**
- Fix classes missing due to inline namespace macro.
- Enable first sentence as brief.
- Make diagrams clickable.

Signed-off-by: Alexander Batashev <alexander.batashev@intel.com>

### CODE REVIEW

File: sycl/doc/doxygen.cfg.in [hunk 1]
```diff
@@ -187,15 +187,15 @@ SHORT_NAMES            = NO
...
+JAVADOC_AUTOBRIEF      = YES
 
 # If the QT_AUTOBRIEF tag is set to YES then doxygen will interpret the first
 # line (until the first dot) of a Qt-style comment as the brief description. If
 # set to NO, the Qt-style will behave just like regular Qt-style comments (thus
 # requiring an explicit \brief command for a brief description.)
 # The default value is: NO.
 
-QT_AUTOBRIEF           = NO
+QT_AUTOBRIEF           = YES
```
Comment (AlexeySachkov) [inline sycl/doc/doxygen.cfg.in]: Why do we need both `QT_AUTOBRIEF` and `JAVADOC_AUTOBRIEF`? I suggest to stick with one style of doxygen comments and therefore, don't encourage usage of others by disabling different helper features for them

Comment (alexbatashev) [reply sycl/doc/doxygen.cfg.in]: @AlexeySachkov is it ok if I prepare a separate pull request that makes the whole documentation up to LLVM standards and disables all Java-style features for good?

Comment (AlexeySachkov) [reply sycl/doc/doxygen.cfg.in]: @alexbatashev, works for me

Comment (bader) [reply sycl/doc/doxygen.cfg.in]: @AlexeySachkov, please, "resolve" conversations to let me know that PRs are ready for merge.

**Pull request merged**